invenio_records_resources/services/events/handlers.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Event handlers module."""
+
+from dataclasses import asdict
+from datetime import datetime
+
+from celery import shared_task
+from flask import current_app
+
+from .bus import EventBus
+
+
+def _handlers_for_key(key):
+    """Returns the handlers for a key."""
+    config_handlers = current_app.config["RECORDS_RESOURCES_EVENTS_HANDLERS"]
+    keys_parts = key.split(".")
+
+    event_handlers = []
+    curr_key = ""
+    for part in keys_parts:
+        curr_key = f"{curr_key}.{part}"
+        try:
+            event_handlers.expand(config_handlers[curr_key])
+        except KeyError:
+            current_app.logger.warning(f"No handler for key {curr_key}")
+
+    return event_handlers
+
+
+def _handle_event(event, handler=None):
+    """Executes the handlers configured for an event."""
+    handlers = _handlers_for_key(event.handling_key)
+
+    for handler in handlers:
+        func = handler
+        async_ = True
+        if isinstance(handler, tuple):
+            func = handler[0]
+            async_ = handler[1]
+
+        if async_:
+            func.delay(**asdict(event))
```

Comment:
We may want to pass more params to the `delay` func: wouldn't it be better to always have a sync func to handle the event, and that it eventually spawns an async task?
```
def hander(...):
    my_async_handler.delay(....)
```